### PR TITLE
Full search replace

### PR DESCRIPTION
**Pull request by hangler**
Adds two new functions:

* `replace_all_text()` will search-and-replace text (using `gsub`, and accepting optional arguments to that function) in either the entire document or the paragraph at the current cursor. Optionally, it will merge runs as well.
* `docx_show_chunk()` will show how the paragraph at the current cursor has been chunked into runs, and the text in each run.

### CODE REVIEW

**Comment by hangler** [comment]
Oops, posted too soon. Got an `rhub` error on my example for `docx_show_chunk`. Fixing that...

**Comment by hangler** [comment]
OK, that fixed it. I get other warnings/errors but they don't seem related to the code I added. I can forward them to you, if you'd like!

**Comment by rodrigomurta** [comment]
@hangler this will be very very useful.
Thanks a lot for you time.

File: DESCRIPTION [hunk 1]
```diff
@@ -3,8 +3,8 @@ Type: Package
 Title: Manipulation of Microsoft Word and PowerPoint Documents
 Version: 0.2.0.0001
 Authors@R: c(
-    person("David", "Gohel", role = c("aut", "cre"),
-    email = "david.gohel@ardata.fr"))
+    person("David", "Gohel", role = c("aut", "cre"), email = "david.gohel@ardata.fr"),
+    person("Frank", "Hangler", role = c("aut"), email = "frank@plotandscatter.com"))
```
Comment (davidgohel) [inline DESCRIPTION]: Could you replace `aut` by `ctb`?

**Comment by davidgohel** [comment]
Thanks a lot

I've made my comments in the PR.

File: DESCRIPTION (hunk 1 above)
Comment (hangler) [reply DESCRIPTION]: Yes, thanks — I meant to ask about the best code to use.

**Comment by hangler** [comment]
@davidgohel A couple of other things I meant to ask:

- I did not write any tests. Should I also add some tests for the functionality, based on existing ones?
- It doesn't look like documentation files were generated for the new functions in `/docs/reference`. Should they have been automatically generated? How can I set that up?

Thanks!

**Comment by davidgohel** [comment]
1. As you wish, if you have time :) I can help with the test files, I will have time later this week and it should not take too much time... 
2. Don't worry about that, I will regenerate the doc locally, this is a manual process and I need to open and save 2 docx files because of tables of content.

**Comment by hangler** [comment]
OK, great! I pushed the commit changing the contributor type but it sounds like it's delayed due to Github issues at the moment.

**Comment by davidgohel** [comment]
Hi,

Did you see the other comments?

**Comment by hangler** [comment]
No — the only one I can see is the comment on the `DESCRIPTION`, for some reason...

**Comment by davidgohel** [comment]
And now?

**Comment by hangler** [comment]
Now I can see them all! Will take a look. Thanks!

**Comment by davidgohel** [comment]
I am learning PR... sorry

File: R/docx_document.R [hunk 2]
```diff
@@ -114,6 +114,75 @@ docx_document <- R6Class(
...
+      search_zone_text <- if (onlyAtCursor) "at the cursor." else "in the document."
+
+      message("Trying to replace '", oldValue, "' with '", newValue, "' ", search_zone_text)
+      message(if (mergeRuns) "Merging runs of text." else "Not merging runs of text. This is faster, but can lead to unexpected results!")
+
+      replacement_count <- 0
+
+      base_node <- if (onlyAtCursor) self$get_at_cursor() else self$get()
+
+      if (mergeRuns) { # Merge all runs to join text
```
[the remark this answers is not in the document]
Comment (hangler) [reply R/docx_document.R]: OK, fair enough. I found it to be useful for some additional docs I was working with, but can keep it on my own fork 😉

**Comment by hangler** [comment]
No problem... I am too 😄

**Comment by hangler** [comment]
OK, those commits should address your comments. Let me know if there's anything else I can do!

**Comment by davidgohel** [comment]
(I never said that before...) *drum roll*, thanks it looks good to me

David

**Comment by hangler** [comment]
Thank you! And thanks for all your support, and for writing a great package in the first place!

**Comment by fabianaboldrin** [comment]
Thank you guys for the help. First of all, it is a great package and very usefull, besides that, the new function was something that i was looking for, now i can do better scripts for modifying .docx files in a more flexible way.

**Pull request merged**